src/misc.rs

```diff
@@ -1,26 +1,13 @@
 //! Miscellaneous pthread functions
 
+// The implementation within `newlib` stubs this out *entirely*. It's not possible to use a "syscall".
```

Comment:
Looking upstream, I see `int __attribute__((weak)) sched_yield (void)` - so we should be able to override this with our version, right?